diff --git a/scene/main/node.cpp b/scene/main/node.cpp
index 9ce036616b..6e706700fb 100644
--- a/scene/main/node.cpp
+++ b/scene/main/node.cpp
@@ -211,6 +211,9 @@ void Node::_notification(int p_notification) {
 				Node *child = data.children.last()->value; // begin from the end because its faster and more consistent with creation
 				memdelete(child);
 			}
+// bianp+2: NOTIFICATION_PREDELETE
+extern void delete_gdobj(Node *);
+delete_gdobj(this);
 		} break;
 	}
 }
