diff --git a/scene/main/node.cpp b/scene/main/node.cpp
index 52c1df8110..cc8ba00dc2 100644
--- a/scene/main/node.cpp
+++ b/scene/main/node.cpp
@@ -169,6 +169,10 @@ void Node::_notification(int p_notification) {
 				Node *child = data.children[data.children.size() - 1]; //begin from the end because its faster and more consistent with creation
 				memdelete(child);
 			}
+			// bianp+2
+			extern void delete_gdobj(Node * p_node);
+			delete_gdobj(this);
+
 		} break;
 	}
 }
