diff --git a/scene/main/node.cpp b/scene/main/node.cpp
index 52c1df8110..6082faa5dc 100644
--- a/scene/main/node.cpp
+++ b/scene/main/node.cpp
@@ -164,11 +164,20 @@ void Node::_notification(int p_notification) {
 				data.parent->remove_child(this);
 			}
 
+			// bianp+
+			extern void delete_gdobj(Node * p_node);
+
 			// kill children as cleanly as possible
 			while (data.children.size()) {
 				Node *child = data.children[data.children.size() - 1]; //begin from the end because its faster and more consistent with creation
+
+				// bianp+
+				delete_gdobj(child);
 				memdelete(child);
 			}
+			// bianp+
+			delete_gdobj(this);
+
 		} break;
 	}
 }
